utilDsu

```diff
@@ -0,0 +1,28 @@
+os.loadAPI("M3C/utils")
+
+dsu = {}
+
+dsu["buildData"] = function ()
+  dsu["data"] = {}
+
+  for block, dsuBlk in pairs(dsu["dsus"]) do
+    local items = dsuBlk.getStoredItems()
+    if items ~= nil then
+      dsu["data"][items.display_name] = items.qty
```

Comment:
2-D arrays are scary. Like I said last night, I think you're over-complicating it for yourself. It seems you're trying to force classes into a language that doesn't really use them. If you're fighting against a language to try to do anything, then there's a good chance you should try a different route.